### PR TITLE
KITCHEN OVERHAUL II: Candyman Edition

### PR DESCRIPTION
**Pull request by Iamgoofball**
Adds 28 new Customizable Foods!
Adds a Candy Machine and a Oven!
Makes deep fryers and customizable foods only take food items!
Expands the amount of combinations the chef can make food wise into the millions!

CURRENT LIST AS OF THIS PR OF CUSTOMIZABLE FOODS:
Sandwich, Soup, Burger, Pizza, Spaghetti, Bread, Pie, Cake, Jelly, Donk Pocket, Kebab, Salad, Waffles, Cookie, Cotton Candy, Giant Gummy Bear, Giant Gummy Worm, Flavored Sucker, Gum, Candy Cane, Giant Jelly Bean, Jawbreaker, Donuts, flavored chocolate bars, Flavored Chocolate Coins, Flavored Chocolate Cash, and then if you fuckers manage to find something I forgot to make a customizable for, there's the Fully Custom, which is made by just putting any food item on a plate.

### CODE REVIEW

File: code/game/machinery/kitchen/cereal_maker.dm [hunk 1]
```diff
@@ -14,14 +14,11 @@
 	if(on)
 		user << "<span class='notice'>[src] is already processing, please wait.</span>"
 		return
-	if(istype(I, /obj/item/weapon/grab)||istype(I, /obj/item/tk_grab))
-		user << "<span class='warning'>That isn't going to fit.</span>"
-		return
-	if(istype(I, /obj/item/weapon/reagent_containers/glass/))
-		user << "<span class='warning'>That would probably break [src].</span>"
+	if(!istype(I, /obj/item/weapon/reagent_containers/food/snacks/))
+		user << "<span class='warning'>Budget cuts won't let you put that in there..</span>"
```
Comment (Atskadan) [inline code/game/machinery/kitchen/cereal_maker.dm]: ?

File: code/game/machinery/kitchen/deep_fryer.dm [hunk 2]
```diff
@@ -20,17 +20,11 @@
...
+	if(!istype(I, /obj/item/weapon/reagent_containers/food/snacks/))
+		user << "<span class='warning'>Budget cuts won't let you put that in there..</span>"
 		return
-	if(istype(I, /obj/item/weapon/reagent_containers/glass))
-		user << "<span class='warning'>That would probably break [src].</span>"
-		return
-	if(!user.unEquip(I))
-		user << "<span class='warning'>You cannot deepfry [I].</span>"
+	if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/deepfryholder))
+		user << "<span class='userdanger'>[I] is already deepfried, you vomituous deviant. This attempt has been logged and will be forwarded to the galactic authorities.</span>"
```
Comment (Atskadan) [inline code/game/machinery/kitchen/deep_fryer.dm]: just say something like "You can't deepfry things multiple times," this isnt really indicative of what's going on

File: code/modules/food/customizables.dm [hunk 3]
```diff
@@ -70,6 +76,60 @@
...
+	desc = "Totally jelly."
+	icon_state = "jellycustom"
+	baseicon = "jellycustom"
+	basename = "jelly"
+	add_overlays = 0
+	top = 0
+
+/obj/item/weapon/reagent_containers/food/snacks/customizable/cook/donkpocket
+	name = "donk pocket"
+	desc = "You wanna put a bangin-Oh fuck it."
```
Comment (Atskadan) [inline code/modules/food/customizables.dm]: i am offended

**Comment by Atskadan** [comment]
i fail to see an oven.dm in this pr

**Comment by Iamgoofball** [comment]
Fuck, forgot to commit it.

File: code/game/machinery/kitchen/cereal_maker.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/game/machinery/kitchen/cereal_maker.dm]: It stops non foods from being deep fried.

File: code/modules/food/customizables.dm (hunk 3 above)
Comment (Iamgoofball) [inline code/modules/food/customizables.dm]: ok soccer mom gosh

File: code/game/machinery/kitchen/deep_fryer.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/game/machinery/kitchen/deep_fryer.dm]: will do

File: code/game/machinery/kitchen/cereal_maker.dm (hunk 1 above)
Comment (Atskadan) [inline code/game/machinery/kitchen/cereal_maker.dm]: but its checking to see if it istype /obj/item/weapon/reagent_containers/food/snacks/
so isnt it checking to see if it IS food and then returning the "you cant put that in there" message?

Comment (RemieRichards) [inline code/game/machinery/kitchen/cereal_maker.dm]: Miggles, theres a ! at the beginning of the istype.
that makes it "If it ISN'T a snack"

This is a non issue, Goof's code is fine here.

Comment (Atskadan) [inline code/game/machinery/kitchen/cereal_maker.dm]: oh ok

**Comment by drynwyn** [comment]
Note should be made- this fixes #449 as grabs are not a food.

**Comment by ghost** [comment]
hype as fuck

File: code/game/machinery/kitchen/deep_fryer.dm [hunk 4]
```diff
@@ -20,17 +20,11 @@
...
+	if(!istype(I, /obj/item/weapon/reagent_containers/food/snacks/))
+		user << "<span class='warning'>Budget cuts won't let you put that in there.</span>"
 		return
-	if(istype(I, /obj/item/weapon/reagent_containers/glass))
-		user << "<span class='warning'>That would probably break [src].</span>"
-		return
-	if(!user.unEquip(I))
-		user << "<span class='warning'>You cannot deepfry [I].</span>"
+	if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/deepfryholder))
+		user << "<span class='userdanger'>[I] is already deepfried, you vomituous deviant.</span>"
```
Comment (Atskadan) [inline code/game/machinery/kitchen/deep_fryer.dm]: this wasnt changed

File: code/game/machinery/kitchen/oven.dm [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	for(var/U in food_choices)
+		food_choices.Remove(U)
+	for(var/U in typesof(/obj/item/weapon/reagent_containers/food/snacks/customizable/cook)-(/obj/item/weapon/reagent_containers/food/snacks/customizable/cook))
+		var/obj/item/weapon/reagent_containers/food/snacks/customizable/cook/V = new U
+		src.food_choices += V
+	return
+
+/obj/machinery/cooking/candy
+	name = "candy machine"
+	desc = "Get yer box of deep fried deep fried deep fried deep fried cotton candy cereal sandwich cookies here!"
```
Comment (Atskadan) [inline code/game/machinery/kitchen/oven.dm:71]: pls

File: code/modules/food/customizables.dm [hunk 6]
```diff
@@ -70,6 +83,143 @@
...
+	name = "flavored sucker"
+	desc = "Suck suck suck."
+	icon_state = "suckercustom"
+	baseicon = "suckercustom"
+	basename = "flavored sucker"
+	add_overlays = 0
+	top = 0
+
+/obj/item/weapon/reagent_containers/food/snacks/customizable/candy/cash
+	name = "flavored cash"
```
Comment (Atskadan) [inline code/modules/food/customizables.dm]: >flavored cash

Comment (Iamgoofball) [inline code/modules/food/customizables.dm]: Think chocolate coins

File: code/game/machinery/kitchen/deep_fryer.dm (hunk 4 above)
Comment (Iamgoofball) [inline code/game/machinery/kitchen/deep_fryer.dm]: Whoops, will fix

File: code/game/machinery/kitchen/oven.dm (hunk 5 above)
Comment (Iamgoofball) [inline code/game/machinery/kitchen/oven.dm:71]: I don't see the problem here but alright.

File: code/modules/food/customizables.dm (hunk 6 above)
Comment (Atskadan) [inline code/modules/food/customizables.dm]: but you have flavored coins just under that

File: code/game/machinery/kitchen/oven.dm (hunk 5 above)
Comment (Atskadan) [inline code/game/machinery/kitchen/oven.dm:71]: there is no problem i was just making a comment its a funny reference

File: code/modules/food/customizables.dm (hunk 6 above)
Comment (Iamgoofball) [inline code/modules/food/customizables.dm]: Some candy store nearby sells fake 1 dollar bills made out of chocolate. I got the idea from that.

**Comment by Atskadan** [comment]
we cant auto this pr blah blah

**Comment by Iamgoofball** [comment]
fukkkkkk

**Pull request merged**